### PR TITLE
Fix OpenWrt SNMP memory plugin

### PR DESCRIPTION
**Pull request by Dismounted**
Basically a rewrite of the existing plugin:
- Added documentation
- Updated to use Munin::Plugin::SNMP
- Fixed OID for snmpd on latest builds of OpenWrt/LEDE

### CODE REVIEW

File: plugins/openwrt/snmp__memory_openwrt [hunk 1]
```diff
@@ -1,76 +1,81 @@
...
-if ($host eq '' || !defined $host)
-{
-	print "# Debug: $0 -- $1\n" if $DEBUG;
-	die "# Error: couldn't understand what I'm supposed to monitor.";
-}
+As a rule SNMP plugins need site specific configuration.  The default
+configuration (shown here) will only work on insecure sites/devices.
+
+   [snmp_*]
+	env.version 2
```
Comment (sumpfralle) [inline plugins/openwrt/snmp__memory_openwrt]: I think, you mixed tabs/spaces here.

File: plugins/openwrt/snmp__memory_openwrt [hunk 2]
```diff
@@ -1,76 +1,81 @@
...
-# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
-#
-#
-# $Log$
-#
-# Inspired by snmp__processes
-#
-#%# family=snmpauto
-#%# capabilities=snmpconf
+#!/usr/local/bin/perl -w
```
Comment (sumpfralle) [inline plugins/openwrt/snmp__memory_openwrt]: I am not aware of the specific situation of perl in openwrt, but I guess, it is installed in `/usr/bin/perl`?
Or is there a good reason to change this path?

File: plugins/openwrt/snmp__memory_openwrt [hunk 3]
```diff
@@ -1,76 +1,81 @@
...
+reports the contents of the hrStorageSize and hrStorageUsed OIDs.
+
+=head1 MAGIC MARKERS
+
+  #%# family=snmpauto
+  #%# capabilities=snmpconf
+
+=head1 VERSION
+
+  $Id$
```
Comment (sumpfralle) [inline plugins/openwrt/snmp__memory_openwrt]: I think, you should put a specifc version here - this is probably a remainder of former svn usage.

File: plugins/openwrt/snmp__memory_openwrt (hunk 1 above)
Comment (Dismounted) [inline plugins/openwrt/snmp__memory_openwrt]: Thanks, will fix.

File: plugins/openwrt/snmp__memory_openwrt (hunk 3 above)
Comment (Dismounted) [inline plugins/openwrt/snmp__memory_openwrt]: I had a look at the newest SNMP plugins included in base munin - they no longer contain this item, so I will do the same.

File: plugins/openwrt/snmp__memory_openwrt (hunk 2 above)
Comment (Dismounted) [inline plugins/openwrt/snmp__memory_openwrt]: Missed this - yep, I have a slightly different environment. Will add a fix for portability.

**Comment by Dismounted** [comment]
@sumpfralle addressed all of you comments - thanks!

**Comment by sumpfralle** [comment]
Great - thank you!

**Pull request merged**